integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java

```diff
@@ -47,14 +46,21 @@
 import io.restassured.RestAssured;
 
 /**
+ * Tests the quarkus:test mojo.
+ *
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
  *         <p>
  *         NOTE to anyone diagnosing failures in this test, to run a single method use:
  *         <p>
  *         mvn install -Dit.test=DevMojoIT#methodName
  */
 @DisableForNative
-public class DevMojoIT extends RunAndCheckMojoTestBase {
+public class DevMojoIT extends LaunchMojoTestBase {
```

Comment:
Looking at this change I had a thought of extracting the CT-related tests from `DevMojoIT` into its own `DevMojoContinuousTestingIT` or something. Then we'd have that and `TestMojoIT` look more consistent and perhaps make it more clear that when new CT tests are to be added they should not go into `DevMojoIT`. WDYT @holly-cummins?

Reply:
Ooh, interesting. 

I can see a few downsides: 
- This one is totally personal taste, but I like to mirror the structure of the classes in the tests, so since we have a `DevMojo` and a `TestMojo`, I'd expect all the tests for each of those to be in `DevMojoIT` and `TestMojoIT` rather having two classes for dev mojo tests. Having said that, I've already violated the 'mirror the class hierarchy' rule by having a common parent just for `DevMojoIT` and `TestMojoIT` even though the dev mojo and test mojo don't share a unique parent.
- It might increase confusion if people have to decide between adding a new class to `DevMojoContinuousTestingIT`, `DevMojoIT,` and the CT superclass.
- Possibly a better reason to keep everything together is that many tests might want to test both application behaviour and CT behaviour. You can see this in `testThatTheApplicationIsReloadedMultiModule` test of [https://github.com/quarkusio/quarkus/blob/main/integration-tests/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java](`DevMojoIT`), where it makes application changes and then interleaves assertions about what's going on in the endpoints with assertions about what's going on in the tests. 

The sort-of counter-argument to this is that this kind of testing is impossible for the test mojo, so I had to duplicate that test and rip out all the endpoint assertions. So although the fact that some dev mojo tests would want to test both app and tests is a problem if we split things out, it's also a problem with how things are now. 

And as you say, there are reasons for: 
- `DevMojoIT` is long, so some subdivision would be valuable
- People might not realise they should be adding their CT tests to the superclass. If a shorter `DevMojoContinuousTestingIT` it would - if nothing else - be easier to spot a big comment saying tests should go in the superclass
- Test apps can still be shared between the two classes, even if they're split up

So I think I'm neutral, overall.